internal/gmm/handler.go

```diff
@@ -1226,6 +1236,7 @@ func handleRequestedNssai(ue *context.AmfUe, anType models.AccessType) error {
 				// Condition (B) Step 7: initial AMF can not find Target AMF via NRF -> Send Reroute NAS Request to RAN
 				allowedNssaiNgap := ngapConvert.AllowedNssaiToNgap(ue.AllowedNssai[anType])
 				ngap_message.SendRerouteNasRequest(ue, anType, nil, ue.RanUe[anType].InitialUEMessage, &allowedNssaiNgap)
+				return err
```

Comment:
Should this be also considered in line 1234 for `callback.SendN1MessageNotifyAtAMFReAllocation(ue, n1Message.Bytes(), &registerContext)`? Otherwise, will RegAceept be sent by the current AMF even when it requests for AMF Re-allocation?

Reply:
Isn't it what the [line 1241](https://github.com/Leon777-coder/amf/blob/735f79503009b51c9141cf5207facb332085cf8e/internal/gmm/handler.go#L1242) doing?

Whenever this handler reach the `needSliceSelection` state (which goes all the way down to the line 1234 if no error occurs), it will `return nil` there (the line 1241). No further operation will be executed. Not to mention `RegAccept` in this situation.